src/ai.c

```diff
@@ -2628,10 +2629,10 @@ static int predict_action_player(game *g, int who, int n, int *leader)
 	player *p_ptr;
 	card *c_ptr;
 	power *o_ptr;
-	int i, x, good[6], count, count_dev, count_world, explore_mix = 0;
+	int i, x, good[MAX_GOOD], count, count_dev, count_world, explore_mix = 0;
 
 	/* Clear good types */
-	for (i = 0; i <= GOOD_ALIEN; i++) good[i] = 0;
+	for (i = 0; i <= MAX_GOOD; i++) good[i] = 0;
```

Comment:
Probably not. All other variables around 'good' are initialized again after this line. I got an error in Windows upon method return, "stack around 'good' was corrupted". You can experiment a little (rerun the smallest network to check for changes), but I guess it will be fine.